extensions/notebook/src/test/book/book.test.ts

```diff
@@ -12,91 +12,140 @@ import * as rimraf from 'rimraf';
+		let folder: vscode.WorkspaceFolder = {
+			uri: vscode.Uri.file(rootFolderPath),
+			name: '',
+			index: 0
+		};
+		bookTreeViewProvider = new BookTreeViewProvider([folder], mockExtensionContext.object);
+		let tocRead = new Promise((resolve, reject) => bookTreeViewProvider.onReadAllTOCFiles(() => resolve()));
+		let errorCase = new Promise((resolve, reject) => setTimeout(() => resolve(), 150));
+		await Promise.race([tocRead, errorCase.then(() => { throw new Error('Table of Contents were not ready in time'); })]);
 	});
 
 	it('should return all book nodes when element is undefined', async function (): Promise<void> {
-		bookTreeViewProvider.onReadAllTOCFiles(async () => {
-			const children = await bookTreeViewProvider.getChildren();
-			should(children).be.Array();
-			should(children.length).equal(1);
-			book = children[0];
-			should(book.title).equal(expectedBook.title);
-		});
+		const children = await bookTreeViewProvider.getChildren();
+		should(children).be.Array();
+		should(children.length).equal(1);
+		book = children[0];
+		should(book.title).equal(expectedBook.title);
 	});
 
 	it('should return all page nodes when element is a book', async function (): Promise<void> {
-		bookTreeViewProvider.onReadAllTOCFiles(async () => {
-				const children = await bookTreeViewProvider.getChildren(book);
-				should(children).be.Array();
-				should(children.length).equal(3);
-				const notebook = children[0];
-				const markdown = children[1];
-				const externalLink = children[2];
-				should(notebook.title).equal(expectedNotebook.title);
-				should(notebook.uri).equal(expectedNotebook.url);
-				should(markdown.title).equal(expectedMarkdown.title);
-				should(markdown.uri).equal(expectedMarkdown.url);
-				should(externalLink.title).equal(expectedExternalLink.title);
-				should(externalLink.uri).equal(expectedExternalLink.url);
-		});
+		const children = await bookTreeViewProvider.getChildren(book);
+		should(children).be.Array();
+		should(children.length).equal(3);
+		notebook1 = children[0];
+		const markdown = children[1];
+		const externalLink = children[2];
+		should(notebook1.title).equal(expectedNotebook1.title);
+		should(notebook1.uri).equal(expectedNotebook1.url);
+		should(notebook1.previousUri).equal(expectedNotebook1.previousUri);
+		should(notebook1.nextUri).equal(expectedNotebook1.nextUri);
+		should(markdown.title).equal(expectedMarkdown.title);
+		should(markdown.uri).equal(expectedMarkdown.url);
+		should(externalLink.title).equal(expectedExternalLink.title);
+		should(externalLink.uri).equal(expectedExternalLink.url);
+	});
+
+	it('should return all sections when element is a notebook', async function (): Promise<void> {
+		const children = await bookTreeViewProvider.getChildren(notebook1);
+		should(children).be.Array();
+		should(children.length).equal(2);
+		const notebook2 = children[0];
+		const notebook3 = children[1];
+		should(notebook2.title).equal(expectedNotebook2.title);
```

Comment:
instead of doing this a number of times, how about having a method that just checks if the property values are equivalent for the 2 objects?

Reply:
created new method for comparing book objects